src/Admin/ReaderThemes.php

```diff
@@ -210,17 +227,38 @@ public function get_default_reader_themes() {
 				]
 			);
 
-			if ( ! is_wp_error( $response ) ) {
-				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
 			}
-		}
 
-		if ( is_wp_error( $response ) ) {
-			return [];
-		}
+			/**
+			 * The response must minimally be an object with a themes array.
+			 *
+			 * @see https://wordpress.org/support/topic/issue-during-activating-the-updated-plugins/#post-13383737
+			 */
+			if (
+				is_wp_error( $response )
+				|| ! is_object( $response )
+				|| ! property_exists( $response, 'themes' )
+				|| ! is_array( $response->themes )
+			) {
+				$this->themes_api_error = new WP_Error(
+					'amp_themes_api_invalid_response',
+					__( 'The request for reader themes from the WordPress.org resulted in an invalid response. Please try again later or contact your host.', 'amp' )
+				);
+				return [];
+			} else {
+				if ( empty( $response->themes ) ) {
+					$this->themes_api_error = new WP_Error(
+						'amp_themes_api_invalid_response',
```

Comment:
This needs a different ID, maybe something like this:
```suggestion
						'amp_themes_api_broken_plugin_override',
```

Reply:
Went with `amp_themes_api_empty_themes_array` in 89485bf0d